### DISCUSSION TITLE
B2B SaaS Next.js Starter Template

### DISCUSSION THREAD

**Discussion by rogadev** [clerk/javascript#4831] [Feature Requests]
I love that you have a simple starter template for using Clerk with Next.js, but the docs actually show this exact template and how to create it, so it's rather redundant.

What I'd really love to see is how `organizations` are meant to work with Clerk and Next.js. The docs _talk_ about organizations but they don't do a very good job of actually showing us how this would/should look in a real Next.js application. If I search the docs organizations page for "onboarding", for example, at no point does it describe an onboarding process. I'd also love to know if/how I can disable "personal" accounts and require users to build one or more organizations, and if/how I can tier a subscription to allow 5/10/25/50 members (for example). Admittedly, I am a junior developer and there might be plenty that is self explanatory, but you've done such a good job explaining other concepts that I'd love to see how Clerk works in this slightly more complicated scenario. I can take the rest, from there.

Would it be possible for someone to create a super simple example of a B2B SaaS Next.js web application that leverages organizations, shows how to handle multi-tenant routing with Clerk middleware, and a basic invitation page? Bonus points if you use this template to update the [docs](https://clerk.com/docs/organizations/overview) as well with some of these example snippets for added clarity.

I really appreciate your platform and I've been using it in every little project that I've been building for myself. Truly, keep up the good work and thank you for providing such an awesome tool. I'll be a happy, paying user as soon as I can figure out this multi-tenant thing. 😅

**Comment by Girolino** [comment]
There is a [demo](https://github.com/clerk/organizations-demo/tree/main) available in Clerk's docs
